### PR TITLE
feat(components): add $bemModifiers in Vuetils Plugin

### PR DESCRIPTION
**Pull request by Slashgear**
- Setup Vuetils Plugin
- add `$bemModifiers` in Plugin
- tests
- Doc

### CODE REVIEW

File: src/__tests__/components.spec.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      {
+        "block__element--visible": true,
+        "block__element--touchable": true,
+        "block__element--is-ghost": true,
+        "block__element--primary": false
+      }
+    ]);
+  });
+
+  it("should be define as prototype function", () => {
```
Comment (Osirisxxl) [inline src/__tests__/components.spec.js]: `should be defined ...` 🇬🇧

File: docs/api/components/bemModifiers.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+## bemModifiers
+
+Returns `Array` with class generator for vue with [BEM Semantic](http://getbem.com/introduction/)
```
Comment (Osirisxxl) [inline docs/api/components/bemModifiers.md]: `BEM semantics` 🇬🇧

File: docs/api/components/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Vue components
+
+Creating components could be verbose sometimes. Those are usefull method for dealing with meta patterns on component creation.
```
Comment (Osirisxxl) [inline docs/api/components/README.md]: `Creating components can sometimes be verbose. Here are some useful methods to handle meta patterns when creating components` 🇬🇧

File: docs/api/components/bemModifiers.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+- **Returns:** `[ bockOrElementDefiniton, { ...modifiers: Boolean }]`
```
Comment (Osirisxxl) [inline docs/api/components/bemModifiers.md]: I don't get what will be returned from reading this 😕

File: src/components.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const toKebabCase = string =>
+  string
+    .replace(/([a-z])([A-Z])/g, "$1-$2")
+    .replace(/\s+/g, "-")
+    .toLowerCase();
+
+export const bemModifiers = function(blockOrElementName, modifiers = {}) {
```
Comment (Osirisxxl) [inline src/components.js]: Can't you use arrow function here ❓ 
```
export const bemModifiers = (blockOrElementName, modifiers = {}) => [
  blockOrElementName,
  ...
];
```

File: src/components.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+const toKebabCase = string =>
+  string
+    .replace(/([a-z])([A-Z])/g, "$1-$2")
+    .replace(/\s+/g, "-")
+    .toLowerCase();
```
Comment (Osirisxxl) [inline src/components.js]: I trust you 😄

File: src/index.js [hunk 7]
```diff
@@ -1 +1,9 @@
 export { withBeforeAfter } from "./methods.js";
+import { bemModifiers } from "./components.js";
+export { bemModifiers } from "./components.js";
```
Comment (Osirisxxl) [inline src/index.js]: If you just export it, isn't it defined in your file anyway ?
Like this:
```
export { bemModifiers } from './components.js';

console.log(bemModifiers); // no reference error 🤞
```

Comment (Slashgear) [reply src/index.js]: Sad but true

**Pull request merged**